document/travis-deploy.sh

```diff
@@ -13,13 +13,18 @@ function doCompile {
   make
   cd ../web-api
   make
+  cd ../core
+  make html
   cd ../../out
   if [[ ! -e js-api ]]; then mkdir js-api; fi
   mv ../document/js-api/_build/html/index.html js-api/index.html
   git add js-api/index.html
   if [[ ! -e web-api ]]; then mkdir web-api; fi
   mv ../document/web-api/_build/html/index.html web-api/index.html
   git add web-api/index.html
+  if [[ ! -e core ]]; then mkdir core; fi
```

Comment:
I'm increasingly confused about this script. The entire doCompile function essentially duplicates the existing Makefile, and in a non-modular manner. Can't we avoid that? The script also shouldn't need to git add things individually.

The existing Makefile creates one _build directory that contains the entire updated contents of the gh-pages branch and then invokes the existing deploy script to update the branch in a single atomic step.

Is there a reason why we cannot simply invoke `make publish` with the right variables instead of replicating everything manually in this script?

Reply:
Deploying this way requires less account privileges than `make publish` does by the use of the ssh key. See https://gist.github.com/domenic/ec8b0fc8ab45f39403dd#get-encrypted-credentials for details. I changed the script here to use `make`; to reduce duplication, maybe we can remove the deployment parts of the Makefile and `deploy.sh` and leave that to this script, unless you still want to do manual deploys sometimes (which aren't so easy with this script)

Reply:
I see, IIUC, this needs checking out as git@github to work? Still would like to unify the two scripts (and keep the ability to manually deploy, especially useful for code reviews on forks). But that isn't high priority -- the make change was the more important bit.